_posts/2021-02-01-Bootable-jar-jkube-clustering-openshift.adoc

```diff
@@ -0,0 +1,425 @@
+---
+layout: post
+date:   2021-02-01
+tags: wildfly galleon bootable jar bootable-jar clustering jkube
+author: yborgess
+---
+
+= WildFly Bootable JAR cluster application with JKube openshift-maven-plugin
+
+== Introduction
+
+This post is a step-by-step guide describing how you can build and deploy on OpenShift an example of a WildFly Bootable JAR application that caches the HTTP session state. We will explore how Bootable JAR uses the http://www.jgroups.org/manual5/index.html#_kube_ping[KUBE_PING] protocol for clustering discovery mechanisms and how you can use https://www.eclipse.org/jkube/docs/openshift-maven-plugin[JKube openshift-maven-plugin] to deploy the application on OpenShift.
+
+== Getting started
+
+The https://github.com/yersan/wildfly-clustering-demo.git[demo application] is a minimalistic shopping cart that stores items in the HTTP session. The key points are:
+
+1. We want our session data to be replicated across all cluster members. The Jakarta Servlet specification supports distributable web applications. If you want to share your session data, you need to specify your session is distributable on the
+_web.xml_ file:
++
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd"
+         version="4.0">
+    <distributable />
+</web-app>
+----
+
+2. Our bootable JAR application is built by using https://docs.wildfly.org/21/Bootable_Guide.html#wildfly_bootable_JAR_development[wildfly-jar-maven-plugin]. This is the default configuration in our _pom.xml_:
++
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#21.0.0.Final</feature-pack-location>
```

Comment:
Should we update your demo to use WildFly 22?

Reply:
Yes, thanks, updated.